packages/adapter/addon/rest.ts

```diff
@@ -1561,91 +1551,4 @@ function ajaxOptions(options: JQueryRequestInit, adapter: RESTAdapter): JQueryRe
   return options;
 }
 
-if (DEPRECATE_NAJAX) {
-  /**
-    @method _najaxRequest
-    @private
-    @param {Object} options jQuery ajax options to be used for the najax request
-  */
-  RESTAdapter.prototype._najaxRequest = function (options: JQueryAjaxSettings): void {
-    if (typeof najax !== 'undefined') {
-      najax(options);
-    } else {
-      throw new Error(
-        'najax does not seem to be defined in your app. Did you override it via `addOrOverrideSandboxGlobals` in the fastboot server?'
-      );
-    }
-  };
-
-  Object.defineProperty(RESTAdapter.prototype, 'useFetch', {
-    get() {
-      if (typeof this[UseFetch] === 'boolean') {
-        return this[UseFetch];
-      }
-
-      // Mixin validates all properties. Might not have it in the container yet
-      let ENV = getOwner(this) ? getOwner(this).resolveRegistration('config:environment') : {};
-      // TODO: https://github.com/emberjs/data/issues/6093
-      let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
-
-      let shouldUseFetch;
-      if (jQueryIntegrationDisabled) {
-        shouldUseFetch = true;
-      } else if (typeof najax !== 'undefined') {
-        if (has('fetch')) {
-          deprecate(
-            'You have ember-fetch and jquery installed. To use ember-fetch instead of najax, set `useFetch = true` in your adapter.  In 4.0, ember-data will default to ember-fetch instead of najax when both ember-fetch and jquery are installed in FastBoot.',
-            false,
-            {
-              id: 'ember-data:najax-fallback',
-              until: '4.0',
-              for: '@ember-data/adapter',
-              since: {
-                available: '3.22',
-                enabled: '3.22',
-              },
-            }
-          );
-        } else {
-          deprecate(
-            'In 4.0, ember-data will default to ember-fetch instead of najax in FastBoot.  It is recommended that you install ember-fetch or similar fetch polyfill in FastBoot and set `useFetch = true` in your adapter.',
-            false,
-            {
-              id: 'ember-data:najax-fallback',
-              until: '4.0',
-              for: '@ember-data/adapter',
-              since: {
-                available: '3.22',
-                enabled: '3.22',
-              },
-            }
-          );
-        }
-
-        shouldUseFetch = false;
-      } else if (hasJQuery) {
-        shouldUseFetch = false;
-      } else {
-        shouldUseFetch = true;
-      }
-
-      addSymbol(this, UseFetch, shouldUseFetch);
-
-      return shouldUseFetch;
-    },
-
-    set(value) {
-      addSymbol(this, UseFetch, value);
-      return value;
-    },
-  });
-} else {
-  Object.defineProperty(RESTAdapter.prototype, 'useFetch', {
```

Comment:
I believe this defineProperty call might have been in use to fix a native-vs-classic class syntax interop issue as much as it was here for the najax fallback. May be worth checking with a test that a consumer can configure this via `extends` vs with `=` vs constructor initialization.